### ISSUE TITLE
2.2 test period

### ISSUE THREAD

**Issue by Ambrevar** [atlas-engineer/nyxt#1681]
I've updated the https://github.com/atlas-engineer/nyxt/tree/2-series branch.

We can target next week for the 2.2 release.
Everyone is welcome to try out the `2-series` branch and report outstanding bugs.

@jmercouris @aartaka Use it as your default Nyxt if you can :)

**Comment by Ambrevar** [comment]
@aartaka wants to include 33212652f596bf58d77fd798f661afebd528faa7.

As of 51e4a0fc870e65c664804a6f99e409d20e044e5a, `master` has the following significant changes:

- b665471bb7829f5e0f4092340a037dbfa82b43c2: history: Fix dead-lock possibility in restore-history-buffers.
- c5f363f05f80e0be07ac1d4f3bdb16a4e5fd35e6: Add support for GTK clipboard

OK to update `2-series` then?

**Comment by jmercouris** [comment]
Ok!

**Comment by Ambrevar** [comment]
Master contains important fixes (notably for the GHT) but it's not ready yet, so I suggest we wait a little bit.

**Comment by Ambrevar** [comment]
Back to business after a long break :)

I've just updated the changelog on the 2-series branch.
@jmercouris @pdelfino @aartaka Can you confirm it looks good for you?

We can release 2.2 tomorrow if that's OK.

There are a few more bugs that'd be nice to fix.  Say, #1783, #1792, #1757, #1678, #1794, #1755, #1743.

Nothing critical though and I'm personally OK with releasing 2.2.0 now and fixing the rest for 2.2.1.  Thoughts?

**Comment by aartaka** [comment]
Yes, looks alright to me! We've been holding this release off for too long :)

**Comment by jmercouris** [comment]
I also confirmed earlier today with Pierre! :-O

**Comment by Ambrevar** [comment]
Another thing I'd like to do for the remaining 2.2.* releases is backport all the backward compatible changes on master.

**Comment by Ambrevar** [comment]
Released!

**Comment by Ambrevar** [comment]
Also please test 2-series for at least a few minutes :)

**Comment by Ambrevar** [comment]
There are still many more bug fixes for the 2.2 series! ;)
  See https://github.com/atlas-engineer/nyxt/issues/1681#issuecomment-940834785.

**Comment by Ambrevar** [comment]
I'd like to release 2.2.3 tomorrow after merging the fixes on master and https://github.com/atlas-engineer/nyxt/pull/1869.

OK with everyone?
@jmercouris @pdelfino @aartaka

**Comment by aartaka** [comment]
I'd also like to enable `onbeforeunload` prompts (they appear in the case you close the buffer with an unsaved input and ask you whether you really want to close the buffer), as it's not pretty to lose one's input after pressing a wrong keybinding (as described [in this forum post on C-w closing the buffer](https://discourse.atlas.engineer/t/c-w-while-in-a-textbox-makes-me-sad/315/1))

If I won't have a fix by tomorrow, let's leave it for 3.0.

**Comment by Ambrevar** [comment]
No problem, 2.2.4 is coming soon, hopefully with the rest of the issues listed in the original post.

Great idea by the way!

**Comment by jmercouris** [comment]
OK with me!

**Comment by Ambrevar** [comment]
Ready to go!
Have a look at the changelog, test it and let me know if there is anything.

I think we can release tomorrow.
@aartaka @pdelfino (and @jmercouris if you've got the time :p)

**Comment by aartaka** [comment]
I've got a crash some two hours ago with `2-series`. Cannot reproduce it, though.

**Comment by Ambrevar** [comment]
Same here :(
That said, I've had many more crashes with 2.2.2 and before :/

Shall we wait a bit?

**Comment by aartaka** [comment]
Yes, let us wait a debug some more.

**Comment by Ambrevar** [comment]
I've used 2-series for a good while now on 2 systems and it worked very
smoothly.

I suggest we go ahead with 2.2.3 today, we can always iron out potential
crashes for 2.2.4.  At any rate, 2.2.3 will be much more stable then
2.2.1 and 2.2.2! :)

**Comment by aartaka** [comment]
> I've used 2-series for a good while now on 2 systems and it worked very smoothly. I suggest we go ahead with 2.2.3 today, we can always iron out potential crashes for 2.2.4. At any rate, 2.2.3 will be much more stable then 2.2.1 and 2.2.2! :)

I've got one more crash yesterday, but it was mostly smooth too. The crashes I got could've been cache problems from WebKit, after all.

Indeed, let's release.

**Comment by Ambrevar** [comment]
One crash on my side is when deleting buffers, in particular too many
buffers at once.  I believe this is related to #1546.  And anyways, it's
been a bug ages, so nothing to stop this release I believe.

**Comment by Ambrevar** [comment]
2.2.3 release!

**Comment by Ambrevar** [comment]
2.2.4 is on its way!
I'd like to merge #1987 at least and maybe implement the history rotation as discussed in #1914.

Please test it, have a look at the commits, the changelog, etc.
@pdelfino @aartaka @guojing0 @jmercouris

**Comment by aartaka** [comment]
@Ambrevar, as embarassing as it is, my fix for iframe redirection wasn't complete :)

Please add aace440910783af8c2c0b5afebfc759647153800 to `2-series`, as it finally makes iframe redirection to not redirect the whole page.f

**Comment by Ambrevar** [comment]
No worries, done!

**Comment by Ambrevar** [comment]
@pdelfino @guojing0 @jmercouris Can you please test `2-series`?  It would be nice to release this week! :)

**Comment by jmercouris** [comment]
Yes, I will be running it!

**Comment by jmercouris** [comment]
As in, I am running it now, nothing to report yet.

**Comment by Ambrevar** [comment]
Thanks @jmercouris !
I've just released 2.2.4!

**Comment by Ambrevar** [comment]
We are probably done here with 2 series.  All issues that are still open in the original post are issues related to the `data-path` library which needs a rewrite, becoming incompatible and thus will be kept for 3.0.

All but one: #1604, this one could still find a spot on 2.2.5.

**Comment by guojing0** [comment]
I thought after 2.2.4 it's 3.0.0

**Comment by Ambrevar** [comment]
Yup, I meant "_if_ someone wants to fix panel input"...
May not happen soon ;)

**Comment by Ambrevar** [comment]
I had to force-push the last 2.2.4 commit because of a mistake in the release.yaml file.  Now fixed!

**Comment by Ambrevar** [comment]
Closing now that 3.0 is imminent.
